### PR TITLE
Refactor memcache module and clean up redis / file cache logic

### PR DESCRIPTION
**Pull request by shifucun**
Also created Redis cache for eval tool model prediction caching (not for other flask view functions).

### CODE REVIEW

**Review by chejennifer — COMMENTED** [review]
nice refactor!

File: server/lib/cache.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+import os
+from pathlib import Path
+
+from flask_caching import Cache
+
+import server.lib.config as lib_config
+import server.lib.redis as lib_redis
+
+_redis_cache = None
```
[the remark this answers is not in the document]
Comment (shifucun) [reply server/lib/cache.py:25]: Added comments

File: server/lib/redis.py [hunk 2]
```diff
@@ -15,35 +15,24 @@
...
-              'CACHE_TYPE': 'RedisCache',
-              'CACHE_REDIS_HOST': host,
-              'CACHE_REDIS_PORT': port,
-              'CACHE_REDIS_URL': 'redis://{}:{}'.format(host, port)
-          })
-    else:
-      cache = Cache(config={'CACHE_TYPE': 'SimpleCache'})
-else:
-  cache = Cache(config={'CACHE_TYPE': 'SimpleCache'})
+    if not region:
```
[the remark this answers is not in the document]
Comment (shifucun) [reply server/lib/redis.py:34]: good catch

**Review by shifucun — COMMENTED** [review]
thanks for review!

File: server/lib/redis.py [hunk 3]
```diff
@@ -15,35 +15,26 @@
...
+  with open(_REDIS_CONFIG) as f:
     redis = json.load(f)
     metadata_url = "http://metadata.google.internal/computeMetadata/v1/instance/zone"
     metadata_flavor = {'Metadata-Flavor': 'Google'}
     zone = requests.get(metadata_url, headers=metadata_flavor).text
     # zone is in the format of projects/projectnum/zones/zone
     region = '-'.join(zone.split('/')[3].split('-')[0:2])
+    if not region:
+      return None
     if region in redis:
```
Comment (chejennifer) [inline server/lib/redis.py:36]: nit: could just do:

```
if not region or region not in redis:
    return None
host = redis[region]["host"]
...
```

File: server/lib/cache.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+import server.lib.redis as lib_redis
+
+# _redis_cache is a redis cache client when a redis config is available.
+# It will be used by the regular flask "cache" and/or the flask "model_cache".
+_redis_cache = None
+
+# cache is the default flask cache for endpoints.
+cache = None
+
+# model_cache is flask cache for endpoints that invoke vertext models.
```
Comment (chejennifer) [inline server/lib/cache.py:30]: vertext -> vertex?

**Review by chejennifer — APPROVED** [review]
thanks for the changes, just some small nits!

File: server/lib/cache.py (hunk 4 above)
Comment (shifucun) [reply server/lib/cache.py:30]: fixed

File: server/lib/redis.py (hunk 3 above)
Comment (shifucun) [reply server/lib/redis.py:36]: good point!

**Pull request merged**